azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -87,3 +89,23 @@ def get_string_tail_lower_case(val, length):
         return val.lower()
 
     return val[len(val) - length :].lower()
+
+
+def real_or_decimal(frame, col):
```

Comment:
Not yours, but: this can actually change the value of decimals, this is the wrong type to parse to (though changing it would be breaking, we need to think about that)

Reply:
I don't know about a different pandas type that would be a better match

Reply:
I guess changing it now would be breaking. I think when we are at exclusively at pandas 2 we can solve this.